std/math.d

```diff
@@ -5768,67 +5826,159 @@ private:
     // Clear all pending exceptions
     static void clearExceptions() @safe
     {
-        version (IeeeFlagsSupport)
-            resetIeeeFlags();
+        version (X86_Any)
+        {
+            version (IeeeFlagsSupport)
+                resetIeeeFlags();
+            else
+                static assert(false, "Not implemented for this architecture");
+        }
+        else version (ARM_Any)
+        {
+            version (IeeeFlagsSupport)
+                resetIeeeFlags();
+            else
+                static assert(false, "Not implemented for this architecture");
+        }
+        else version (PPC_Any)
+        {
+            asm pure nothrow @nogc @safe
+            {
+                `mtfsb0 24
+                 mtfsb0 25
+                 mtfsb0 26
+                 mtfsb0 27
+                 mtfsb0 28`;
+            }
+        }
+        else version (MIPS_Any)
+        {
+            version (D_LP64) enum mask = 0xFFFFF07Fu;
+            else             enum mask = 0xF07Fu;
+
+            const newState = getControlState() & mask;
+            setControlState(newState);
+        }
         else
             static assert(false, "Not implemented for this architecture");
     }
 
     // Read from the control register
     static ControlState getControlState() @trusted pure
     {
-        version (D_InlineAsm_X86)
+        ControlState cont;
+
+        version (X86)
         {
-            short cont;
-            asm pure nothrow @nogc
+            version (DigitalMars)
+            {
+                asm pure nothrow @nogc
+                {
+                    xor EAX, EAX;
+                    fstcw cont;
```

Comment:
I have no idea why EAX/RAX is set to 0 before `fstcw`. That instruction doesn't seem to depend on it (https://www.felixcloutier.com/x86/fstcw:fnstcw).